tap_mixpanel/schema.py

```diff
@@ -1,125 +1,156 @@
+        schema["additionalProperties"] = True
     else:
-        schema['additionalProperties'] = False
+        schema["additionalProperties"] = False
 
-    if stream_name == 'engage':
+    if stream_name == "engage":
         properties = client.request(
-            method='GET',
-            url='https://{}/api/2.0'.format(client.__api_domain),
-            path='engage/properties',
-            params={'limit': 2000},
-            endpoint='engage_properties')
-        if properties.get('status') == 'ok':
-            results = properties.get('results', {})
+            method="GET",
+            url=f"https://{client.__api_domain}/api/2.0",
+            path="engage/properties",
+            params={"limit": 2000},
+            endpoint="engage_properties",
+        )
+        if properties.get("status") == "ok":
+            results = properties.get("results", {})
             for key, val in results.items():
-                if key[0:1] == '$':
-                    new_key = 'mp_reserved_{}'.format(key[1:])
+                if key[0:1] == "$":
+                    new_key = f"mp_reserved_{key[1:]}"
                 else:
                     new_key = key
 
                 # property_type: string, number, boolean, datetime, object, list
                 # Reference:
                 # https://help.mixpanel.com/hc/en-us/articles/115004547063-Properties-Supported-Data-Types
-                property_type = val.get('type')
+                property_type = val.get("type")
 
                 types = {
-                    'boolean': {
-                        'type': ['null', 'boolean']
-                    },
-                    'number': {
-                        'type': ['null', 'string'],
-                        'format': 'singer.decimal'
-                    },
-                    'datetime': {
-                        'type': ['null', 'string'],
-                        'format': 'date-time'
+                    "boolean": {"type": ["null", "boolean"]},
+                    "number": {"type": ["null", "string"], "format": "singer.decimal"},
+                    "datetime": {"type": ["null", "string"], "format": "date-time"},
+                    "object": {
+                        "type": ["null", "object"],
+                        "additionalProperties": True,
                     },
-                    'object': {
-                        'type': ['null', 'object'],
-                        'additionalProperties': True
-                    },
-                    'list': {
-                        'type': ['null', 'array'],
-                        'required': False,
-                        'items': {}
-                    },
-                    'string': {
-                        'type': ['null', 'string']
-                    }
+                    "list": {"type": ["null", "array"], "required": False, "items": {}},
+                    "string": {"type": ["null", "string"]},
```

Comment:
Compared to this code, I find original code is more readable. Any reason we changed it?

Reply:
re-modified